src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -393,4 +394,12 @@ def validate_and_append_build_arguments(build_arg,
             if "=" not in name_value:
                 raise CLIError("Accepted format for arg is <name>=<value>.")
             name, value = name_value.split('=', 1)
-            build_arguments.append(BuildArgument(name, value, is_secret))
+            if is_template_deployment:
+                build_arguments.append({
```

Comment:
a comment might be helpful here, the serialization from a `BuildArgument` is different than this...? It looks identical to me looking at https://github.com/AzureCR/azure-cli/blob/dev/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/sdk/models/build_argument.py#L33-L36

Reply:
For build-task create, we do a ARM template deployment. I checked the value of build_arguments before it was sent to the template and it was indeed an array of BuildArgument objects. The azure-cli stringifies the BuildArgument object in the array, and hence the build-step creation was failing yesterday. Hence, sending in a hand-crafted object to template like used in docs: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-array#example

This is not the case for quickbuild where we do not do an ARM template deployment, and hence the sdk handles the request fine.